lib/tests/modules/declaration-positions.nix

```
@@ -0,0 +1,42 @@
+{ lib, options, ... }:
+let discardPositions = lib.mapAttrs (k: v: v);
+in
+{
+  imports = [
+    {
+      options.imported.line7 = lib.mkOption {
+        type = lib.types.int;
+      };
+
+      # Simulates various patterns of generating modules such as
+      # programs.firefox.nativeMessagingHosts.ff2mpv. We don't expect to get
+      # line numbers for these, but we can fall back on knowing the file.
+      options.generated = discardPositions {
+        line15 = lib.mkOption {
+          type = lib.types.int;
+        };
```

What if you insert a nested option here? That should work, right?

```suggestion
        };
        nest.nested = lib.mkOption { };
```

---

yes, and test added